uctMaths/competition/compadmin.py

```diff
@@ -1288,25 +1296,121 @@ def get_student_answer_sheet(request, student):
         template = get_template('pair_as_template.html')
     else:
         template = get_template('individual_as_template.html')
-    c.update(csrf(request))
     context = Context(c)
     return template.render(context)
 
+def generate_answer_sheets(request, school_list):
```

Comment:
Can you move this back into views.py please. It will make it easier to see what is happening in the diff.

Reply:
This doesn't work, background_tasks has request as None, which I think isn't allowed in functions in views.py as the traceback ends on the function call into views.py
(This should be here)